docs/specification.rst

```diff
@@ -1,668 +1,481 @@
+possible to mix and match, it is predictable that making a decision either way
+on this would to lead to increased friction between typed and untyped code, and
+increase pressure to more fully type code or to treat code that still can't be
+expressed by the type, but which needed to interact with typed code as taboo.
+
+The ordering allows expressing preferring either the False positives for the
+implied possible diamond pattern with untyped things (``T & Any``) and a narrow
+remedy for it (``P & T & Any``) where ``P`` is a protocol expressing how the
+diamond pattern was actually resolved.
+
+The ordering also allows expressing prefering not to get warnings for the
+implied possible diamond pattern: (``Any & T``)
+
+Neither of these provide warnings for things not provided by ``T``, the scope
+of the ordering mattering is only in the overlap.
+
+(TODO add table comparing effects of each option in each of meaningful cases, highlighting the equivalence to unordered in most cases)
+
+Backwards Compatibility
+=======================
+
+This PEP expands the allowed use of ``Self`` to better handle mixins,
+this change is not done in a backwards incompatible manner.
+
+The implementation of ``__and__`` for the builtin ``type`` may result in
+runtime uses of type introspection to misbehave for user defined types using
+a metaclass which defines ``__and__`` for some purpose.
+
+The considerations are similar to the prior implementing of ``|`` for types.
+
+
+Security Implications
+=====================
+
+None
+
+
+How to Teach This
+=================
+
+At the most universal level, Intersections just express that a value must
+be consistent with all of the types of the Intersection.
+This is a clear parallel to Unions, which express that a value must be
+consistent with one of the types of the Union.
+
+When teaching and documenting the edge cases that involve what happens when
+using a value that involves diamond patterns, we reccoemend drawing a parallel
+to MRO, as well as reccomending that library authors avoid creating an
+expectation that users be the one to solve a diamond pattern created by a
+library when possible. There may always be advanced use-cases where doing so is
+not possible; However, users finding themselves in this situation will need to
+understand MRO to resolve them, and the ordering in use here matches that.
+
+
+Reference Implementation
+========================
+
+TODO
+
+Comparison to other languages
+=============================
+
+When comparing what other languages with intersections, most do not provide an
+intersection type.
+
+Two notable comparisons with languages that do reinforce the decision to
+include ordering.
+
+While Kotlin (unlike Java) allows multiple inheritence, it does not allow a
+diamond pattern to exist, allowing only one base to implement a defintion for
+a method or attribute of a type. Kotlin's intersections therefore do not have
+to consider non-disjoint intersections.

```

Comment:
When not targeting the JVM, Kotlin's `dynamic` is available. Synthesizing bounds for dynamic is not something currently done: https://github.com/JetBrains/kotlin/blob/master/spec-docs/dynamic-types.md you may want to specify Kotlin, while targeting the JVM or reword this to account for dynamic better.

Reply:
I'm not sure we need to bring this part into the discussion. I'll think about it more, but the prior part sticks to why it's the case, rather than an example of how.

it might also be worth adding that while Kotlin has intersection types, it has them through smart casting and they are not directly user-denotable 

They arise with

```kt
fun process(item: Any) {
    if (item is A && item is B) {
        // smart cast to an intersection of A and B
    }
}
```

or (compile time checked that use of the function is as expected)

```kt
fun <T> process(item: T) where T: A, T: B {
    // item is an intersection of A and B
}
```

But expressed as bounds on a type.

Reply:
I added details about kotlin's being non-denotable. let me know if that's enough additional clarity for what you were thinking in that section